### PR TITLE
Use official SQ Runner API to avoid compatibility issues

### PR DESCRIPTION
**Pull request by henryju**
- org.codehaus.sonar.runner:sonar-runner-dist is not an API so even if the ability to choose version of SQ Runner is applealing there is no garantee that future version will remain compatible
- use official entry point ForkedRunner
- remove ability to choose version of SQ Runner

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,6 +19,7 @@ import org.gradle.build.BuildTypes
 
 defaultTasks 'assemble'
 apply plugin: 'java-base'
+apply plugin: 'sonar-runner'
```
Comment (alkemist) [inline build.gradle]: Is there a reason you added this?

**Comment by alkemist** [comment]
Thanks @henryju .

Can you provide some rationale as to why this is necessary. This change has serious implications of forward compatibility and the size of our distribution. Are you saying the that runner version is forward compatible with different server versions? If not, then doesn't this mean that use of this plugin binds people to SonarQube 2.3? That seems like a huge limitation.

It's also not clear why the changes to the property values are being made. Can you provide some explanation on this too?

File: build.gradle (hunk 1 above)
Comment (henryju-test) [inline build.gradle]: My goal was to be able to analyze Gradle itself with sonarRunner task as a proof of concept (eat your own dogfood). But probably I should have proposed this change in a separate pull request. I'll remove this change.

**Comment by henryju-test** [comment]
SonarQube Runner API is the official entry point to run SQ analysis. It's version is not limited to a single SonarQube version. For example SonarQube Runner 2.4 supports SonarQube 3.7+ (as a reminder current LTS is 4.5.x). SonarQube Runner API 2.4 is used by the Maven plugin, the Ant task, Eclipse and IntelliJ integration... The old way is not supported and has in my opinion only the advantage to not embed the 200Kb sonar-runner-api dependency but in turn require to support and document this confusing "sonar runner version" parameter (most users don't understand what it implies, and on our side we generally always ask to use latest version).

Concerning the change of properties generation this is to fix a problem I have detected during my tests. I was trying to analyse Gradle itself with sonarRunner task. The problem was that the module named "sonar" was generating properties with prefix "sonar.". This resulted in having properties of the form:
* sonar.sonar.sources=xxx
* sonar.sonar.tests=yyy
* ...
This duplicate "sonar." prefix resulted in some confusion on SonarQube side where properties were associated to the wrong module (I have created https://jira.codehaus.org/browse/SONAR-6514 on our side to fail with a proper message). 
"sonar" should be considered as a reserved keyword. In order to limit the probability of using this reserved keyword I changed property generation to use "group:module name" as prefix. Tell me if you prefer to have this as a separate P/R.

**Comment by mark-vieira** [comment]
@henryju I'm closing this PR since we have made the decision to deprecate the Sonar plugin in Gradle core in favor of an external plugin maintained by the Sonar folks. The mailing list discussion about this can be found [here](https://groups.google.com/d/msg/gradle-dev/daGh2oy-0kU/5ywN4nfgDYYJ). You can also take a look at the [git repo](https://github.com/SonarCommunity/sonar-gradle) for the plugin to see if this change is still applicable.

**Pull request closed without merging**